utils/report_targets.py

```diff
@@ -174,6 +187,9 @@ def main():
             continue
 
         arch, design, board, stage = fields
+
+        assert (arch, design) in allowed_failures, f"{arch} {design} test is not allowed to fail"
```

Comment:
It may be better to first collect all failed but not allowed to fail tests and then print them out before triggering the CI failure. This way we would get a list of what failed in the log instead of just a single test name.

Reply:
Sure, fixed, a list of failing targets is output in the end before exiting with an error.